### PR TITLE
WIP: Additional environment properties, fix interfaceType and gripperInfo 

### PR DESCRIPTION
**Pull request by bucketzxm**
1. Add name, keywords and description to scene asset properties
2. Fix the re-create robot pointer problem, caused by interfacetype
3. Fix extract and update gripperInfo

### CODE REVIEW

File: python/bindings/openravepy_robot.cpp [hunk 1]
```diff
@@ -482,12 +492,38 @@ std::vector<RobotBase::ConnectedBodyInfoPtr> ExtractConnectedBodyInfoArray(objec
...
+        for(size_t iGripperInfo = 0; iGripperInfo < arraySize; iGripperInfo++) {
+            RobotBase::GripperInfoPtr pInfo(new RobotBase::GripperInfo());
+            rapidjson::Document rGripperInfo;
+            toRapidJSONValue(pyGripperInfoList[iGripperInfo], rGripperInfo, rGripperInfo.GetAllocator());
+            pInfo->DeserializeJSON(rGripperInfo, fUnitScale, options);
+            vGripperInfos.push_back(pInfo);
+        }
+    }
+    catch(...) {
+        RAVELOG_WARN("Cannot do ExtractArray for GripperInfo");
```
Comment (bucketzxm) [inline python/bindings/openravepy_robot.cpp:515]: From Rosen: @bucketzxm i really don't like this type of catch all usage...https://github.com/rdiankov/openrave/pull/872

File: src/libopenrave-core/environment-core.h [hunk 2]
```diff
@@ -2440,18 +2443,26 @@ class Environment : public EnvironmentBase
         // TODO: revision error checking ?
         SetRevision(info._revision);
 
+        // set keywords and description
+        name = info.name;
+        keywords = info.keywords;
+        description = info.description;
+
+        RAVELOG_DEBUG("=== UpdateFromInfo start ===");
```
Comment (rdiankov) [inline src/libopenrave-core/environment-core.h:2451]: @bucketzxm perhaps verbose is better..

File: src/libopenrave-core/environment-core.h [hunk 3]
```diff
@@ -2440,18 +2443,26 @@ class Environment : public EnvironmentBase
...
         FOREACHC(itBodyInfo, info._vBodyInfos) {
             KinBody::KinBodyInfoPtr pKinBodyInfo = *itBodyInfo;
             if(pKinBodyInfo->_id.empty()){
                 pKinBodyInfo->_id = pKinBodyInfo->_name;
             }
+            RAVELOG_DEBUG_FORMAT("==== %s ===", pKinBodyInfo->_id);
```
Comment (rdiankov) [inline src/libopenrave-core/environment-core.h:2457]: @bucketzxm perhaps verbose is better..

**Comment by bucketzxm** [comment]
Merged to `webstack2` branch.

**Pull request closed without merging**